File: app/src/main/java/com/example/madarsofttask/ui/add/AddUserViewModel.java
package com.example.madarsofttask.ui.add;

import androidx.lifecycle.ViewModel;

import com.example.madarsofttask.data.models.User;
import com.example.madarsofttask.data.repositories.UserRepository;

public class AddUserViewModel extends ViewModel {
    private UserRepository userRepo;

    public AddUserViewModel() {
        userRepo = UserRepository.getInstance();
    }

    void addUser(String name, String title, double age, String gender) {
        User user = new User();
        user.setName(name);
        user.setJobTitle(title);
        user.setAge(age);
        user.setGender(gender);
        userRepo.saveUser(user);
    }
}

File: app/src/main/java/com/example/madarsofttask/ui/add/AddUserViewModel.kt
package com.example.madarsofttask.ui.add

import androidx.lifecycle.ViewModel
import com.example.madarsofttask.data.models.User
import com.example.madarsofttask.data.repositories.UserRepository
import com.example.madarsofttask.data.repositories.UserRepository.Companion.instance

class AddUserViewModel : ViewModel() {
    private val userRepo: UserRepository? = instance
    fun addUser(name: String?, title: String?, age: Double, gender: String?) {
        val user = User()
        user.name = name
        user.jobTitle = title
        user.age = age
        user.gender = gender
        userRepo!!.saveUser(user)
    }

}